src/security/Security.js

```diff
@@ -509,11 +509,47 @@ Security.prototype.createUser = function (id, content, options, cb) {
     options = null;
   }
 
-  if (options) {
-    action = options.replaceIfExist ? 'createOrReplaceUser' : 'createUser';
+  if (options && options.replaceIfExist) {
+    self.fetchUser(id, function (fetchError, fetchResult) {
+      if (fetchResult instanceof User && options && options.replaceIfExist === true) {
```

Comment:
We can reject the query here if `fetchUser` return a user and `options.replaceIfExist !== true` without querying kuzzle